src/api.rs

```diff
@@ -1054,7 +1055,15 @@ impl Api {
         // to be a Sentry-compatible endpoint, we force the Authorization header at
         // this point.
         let request = match Config::current().get_auth() {
-            Some(auth) => request.with_auth(auth)?,
+            Some(auth) => {
```

Comment:
You can simplify this:

```suggestion
            Some(auth) if !request.is_authenticated => {
```